src/index.js

```diff
@@ -15,8 +15,12 @@ const INSIDE_DISPATCH_MESSAGE = [
   '(http://redux.js.org/docs/Troubleshooting.html#never-mutate-reducer-arguments)'
 ].join(' ');
 
-export default function immutableStateInvariantMiddleware(isImmutable = isImmutableDefault) {
-  const track = trackForMutations.bind(null, isImmutable);
+export default function immutableStateInvariantMiddleware(
+  isImmutable = isImmutableDefault,
```

Comment:
I think I'd rather make the middleware just receive an options object, and put in there the `ignore` option along with the `isImmutable` function. 

`isImmutable` is not documented and also it's not very common to change it, making it inconvenient as a first argument (making you pass `undefined` as you did un several places).